vllm/model/model.py

```diff
@@ -0,0 +1,204 @@
+        else:
+            try:
+                result = subprocess.run(
+                    ["nvidia-smi"], capture_output=True, text=True, check=True
+                )
+                logger.info(result.stdout)
+            except subprocess.CalledProcessError as e:
+                logger.error(f"Command failed with code {e.returncode}: {e.stderr}")
+
+            self.model_args = AsyncEngineArgs(
+                model=self._model_repo_id, **self._vllm_config
+            )
+            self.llm_engine = AsyncLLMEngine.from_engine_args(self.model_args)
+            self.tokenizer = AutoTokenizer.from_pretrained(self._model_repo_id)
+
+            try:
+                result = subprocess.run(
+                    ["nvidia-smi"], capture_output=True, text=True, check=True
+                )
+                logger.info(result.stdout)
+            except subprocess.CalledProcessError as e:
+                logger.error(f"Command failed with code {e.returncode}: {e.stderr}")
+
+    async def predict(self, model_input):
+        if "messages" not in model_input and "prompt" not in model_input:
+            raise ValueError("Prompt or messages must be provided")
+
+        stream = model_input.pop("stream", False)
+        if self.openai_compatible:
+            # if the key metrics: true is present, let's return the vLLM /metrics endpoint
+            if model_input.get("metrics", False):
+                response = await self._client.get(f"{self.vllm_base_url}/metrics")
+                return response.text
+
+            # convenience for Baseten bridge
+            if "model" not in model_input and self._model_repo_id:
+                logger.info(
+                    f"model_input missing model due to Baseten bridge, using {self._model_repo_id}"
+                )
+                model_input["model"] = self._model_repo_id
+
+            if stream:
+
+                async def generator():
+                    async with self._client.stream(
+                        "POST",
+                        f"{self.vllm_base_url}/v1/chat/completions",
+                        json=model_input,
+                    ) as response:
+                        async for chunk in response.aiter_bytes():
+                            if chunk:
+                                yield chunk
+
+                return generator()
+            else:
+                response = await self._client.post(
+                    f"{self.vllm_base_url}/v1/chat/completions",
+                    json=model_input,
+                )
+                return response.json()
+        else:
+            # SamplingParams does not take/use argument 'model'
+            if "model" in model_input:
+                model_input.pop("model")
+            if "prompt" in model_input:
+                prompt = model_input.pop("prompt")
+                sampling_params = SamplingParams(**model_input)
+                idx = str(uuid.uuid4().hex)
+                messages = [
```

Comment:
do you feel strongly about this? I can also see an argument for setting input = prompt directly, i.e. if the user wants to deal with templating on their side

Reply:
I would prefer not to do that cause I want to make API as simple as possible if it can cover 90% use case. Supporting raw input may need to add an extra flag in the request body like "enable_raw_input".
If we see high customer demand for this use case, I will change my mind.